app/Model/AppModel.php

```diff
@@ -1395,6 +1395,9 @@ public function updateDatabase($command)
             case 54:
                 $sqlArray[] = "ALTER TABLE `sightingdbs` MODIFY `timestamp` int(11) NOT NULL DEFAULT 0;";
                 break;
+            case 55:
+                $sqlArray[] = "ALTER TABLE `correlations` DROP `info`, DROP `date`, DROP INDEX `value`;";
```

Comment:
OK, with this we need to be careful. We have correlation tables with 100s of millions of entries so this will be extremely slow, perhaps if we want to go down this route the best would be to truncate the table, alter the table and initiate a recorrelation. Will think about this a bit more on Monday.

Reply:
Dropping index should be OK ([happens in place and without table rebuild](https://dev.mysql.com/doc/refman/5.6/en/innodb-online-ddl-operations.html)), but yes, dropping columns require table rebuild.

But because recorrelate all attributes can be very slow (much slower that altering table), we can create background job, that will drop columns. Or do not drop them, just don't use them anymore and provide script to drop them manually by MISP admins.

Reply:
It will be slower, but the system remains usable while it is running. We have a background processed system for slow upgrades like that, the only hurdle is that it will create a potentially hour+ long outage on instances for the alter table (we have some larger instances for example that are heavily used and are running on HDDs with massive correlation tables).

So our options are:

1. Truncate, alter table, recorrelate
2. Update system lock, drop indeces, alter table, reindex
3. Keep the fields just stop using them until an admin executes a removal task that will either run 1 or 2.

Reply:
Now I am thinking that the best solution will be recorrelate everything. Because now  there are some fixes that affect correlations:
* port in `ip-src|port` attribute was improperly correlated with other attributes 
* IPv6 CIDR correlations was broken
* Value for CIDR and ssdeep correlation didn't show correlated value
* ssdeep correlation was also broken (#5917)

And because generating correlations should be now faster, I think it makes sense to truncate table, drop columns and run correlation generation.